libs/services/src/dune-sender/dune-sender.service.ts

```diff
@@ -0,0 +1,133 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { Cron, CronExpression } from "@nestjs/schedule";
+import { Lock, OriginLogger } from "@multiversx/sdk-nestjs-common";
+import { DuneClient, ColumnType, ContentType, DuneError } from "@duneanalytics/client-sdk";
+import { CsvRecordsService } from "../records";
+import { AppConfigService } from "apps/api/src/config/app-config.service";
+import axios from 'axios';
+import { CsvFile } from "apps/dune-mock/src/endpoints/dune-mock/entities/csv.file";
+@Injectable()
+export class DuneSenderService {
+    private readonly logger = new OriginLogger(DuneSenderService.name);
+
+    constructor(
+        private readonly csvRecordsService: CsvRecordsService,
+        private readonly appConfigService: AppConfigService,
+    ) { }
+
+    client = new DuneClient(this.appConfigService.getDuneApiKey());
+
+    @Cron(CronExpression.EVERY_30_SECONDS)
+    @Lock({ name: 'send-csv-to-dune', verbose: false })
+    async sendCsvRecordsToDune(): Promise<void> {
+        const records: Record<string, string[]> = this.csvRecordsService.getRecords();
+
+        await this.sendCsvToDune(records);
+    }
+
+    async sendCsvToDune(records: Record<string, string[]>) {
+        for (const [csvFileName, lines] of Object.entries(records)) {
+            if (lines.length === 0) {
+                continue;
+            }
+            const [resultString, linesLength] = await this.csvRecordsService.formatRecord(csvFileName);
+
+            const csvData: Buffer = Buffer.from(resultString, 'utf-8');
+
+            this.logger.log("starting sending data from file " + csvFileName);
+
+            const formattedCsvFileName = csvFileName.toLowerCase().replace(/-/g, "_");
+            const isRecordSent = this.appConfigService.isDuneSendingEnabled() ?
```

Comment:
ideally, we should use the same code for both the simulated or the real dune. In order to to this, I think we should rely to direct API calls instead of using their library. The dune API should be fetched from config and the same code should work for both ways without any `if` inside the code